packages/service-provider-server/src/cli-service-provider.ts

```diff
@@ -236,9 +236,19 @@ class CliServiceProvider extends ServiceProviderCore implements ServiceProvider
   async getConnectionInfo(): Promise<ConnectionInfo> {
     const topology = this.getTopology();
     const { version } = require('../package.json');
-    const [buildInfo = null, cmdLineOpts = null, atlasVersion = null, fcv = null] = await Promise.all([
+    const [buildInfo = null, cmdLineOptsOrServerError = null, atlasVersion = null, fcv = null] = await Promise.all([
       this.runCommandWithCheck('admin', { buildInfo: 1 }, this.baseCmdOptions).catch(() => {}),
-      this.runCommandWithCheck('admin', { getCmdLineOpts: 1 }, this.baseCmdOptions).catch(() => {}),
+      this.runCommandWithCheck('admin', { getCmdLineOpts: 1 }, this.baseCmdOptions).catch((e) => {
+        // mongodb-build-info.getGenuineMongoDB expects either
+        // the successful or failure response from server
+        // Ref: https://github.com/mongodb-js/mongodb-build-info/blob/main/index.js#L89
```

Comment:
I’d make this a permalink since the line numbers might shift around on a given branch:

```suggestion
        // Ref: https://github.com/mongodb-js/mongodb-build-info/blob/9247eeba730a905397ad09fe5a377067edc49b34/index.js#L89
```

Reply:
oh that's right